### DISCUSSION TITLE
Running into an "undefined symbol" issue with the installation

### DISCUSSION THREAD

**Discussion by ybadr16** [neams-th-coe/cardinal#948] [Q&A] [answered]
I tried the installing cardinal without moose's conda environment guide, everything went smoothly with the exception of a few timeouts issues during the cloning phase of the dependencies file (resolved by switching the protocol to ssh in .gitmodules), the installation produced the exe cardinal-opt successfully. However, whenever I run it i get the following:
`mpiexec -np 2 ../../../../../cardinal-opt -i openmc_master.i --n-threads=2`
`../../../../../cardinal-opt: symbol lookup error: /home/youssef/cardinal/lib/libcardinal-opt.so.0: undefined symbol: _ZN6openmc8settings12libmesh_commE`
`../../../../../cardinal-opt: symbol lookup error: /home/youssef/cardinal/lib/libcardinal-opt.so.0: undefined symbol: _ZN6openmc8settings12libmesh_commE`

or for another command 
`./cardinal-opt --help`
`./cardinal-opt: symbol lookup error: /home/youssef/cardinal/lib/libcardinal-opt.so.0: undefined symbol: _ZN6openmc8settings12libmesh_commE`

I have tried re-installing libmesh, which did not seem to work. I am unsure what to do to solve the problem.

**Comment by aprilnovak** [comment]
Hi @ybadr16 , sorry for the delayed response - I was traveling internationally last week so am catching up on emails now.

I think I'll need to see a bit more info to help debug this. Can you please do the following (re-build Cardinal, and save the console output to a file) and attach `out.txt` here:

```
cd cardinal
rm -rf build/ install/
./scripts/get-dependencies.sh

make -j8 MAKEFLAGS=-j8 >> out.txt 2>&1 
```

**Reply by ybadr16** [reply]
That is okay, thank you for your help. I have attached out.txt, and i have tried re-running it but it seems to face the same problem. 
[out.txt](https://github.com/user-attachments/files/16580495/out.txt)

One thing i have noticed is that the openmc version does not work on its own..  Here is the output when i try running an openmc script i have.

[openmc_output.txt](https://github.com/user-attachments/files/16580536/openmc_output.txt)

Note that I am sure that this script runs as i use it consistently when i compile openmc from source. If i remove the cardinal environment variable and use the openmc installed in usr/local/bin it works perfectly.

**Reply by aprilnovak** [reply]
Can you please post the script that you are referring to with regards to running OpenMC, so I can see exactly what you're referring to?

**Reply by ybadr16** [reply]
This is the script. 
One more thing i discovered is that it is running if i run it through python, but if i run it with openmc targeting settings.xml i get the output i provided above.

(I changed the format to .txt because github comments do not support .py).
[openmctest (copy).txt](https://github.com/user-attachments/files/16581280/openmctest.copy.txt)

**Reply by pshriwise** [reply]
I'm late to the party here, but @ybadr16 could you post the output of the following command:

```shell
$ ldd /home/youssef/cardinal/lib/libcardinal-opt.so.0
```

**Reply by ybadr16** [reply]
Here is the output
[libraries.txt](https://github.com/user-attachments/files/16590695/libraries.txt)

**Reply by pshriwise** [reply, marked as answer]
This line is jumping out at me:

```
libopenmc.so => /usr/local/lib/libopenmc.so (0x00007b6cb6c00000)
```

And doesn't appear to be the OpenMC library that is generated by a Cardinal build. This is most likely why that symbol is declared missing by the `libcardinal-opt` library at runtime.

Can you provide the output of

```shell
$ env
```

as well?

**Reply by ybadr16** [reply]
This is possibly because I have another install from source for openmc?

Here is the env output 
[env.txt](https://github.com/user-attachments/files/16591033/env.txt)

**Reply by ybadr16** [reply]
Yeah that was it. Unsure how to resolve the conflict, but moving libopenmc.so from /usr/local/lib resolved the issue and now cardinal works fine. Any ideas on how can i keep both without causing any conflict?

**Reply by ybadr16** [reply]
Okay so, I added 
`$ export LD_LIBRARY_PATH=/home/youssef/cardinal/install/lib:$LD_LIBRARY_PATH`
to my bash_profile. Whenever i want to use cardinal i just source it. This resolves the issue.

**Reply by pshriwise** [reply]
That would do the trick, yep. I generally recommend trying to avoid setting the `LD_LIBRARY_PATH` variable at all b/c it can override the paths to other libraries embedded in the `.so` files, but I know that's not always possible and if this keep you moving, great!

**Reply by ybadr16** [reply]
Thank you all for your help!
